client/dashboard/app/queries/site.ts

```diff
@@ -26,6 +26,23 @@ export const siteBySlugQuery = ( siteSlug: string ) => ( {
 } );
 
 export const siteByIdQuery = ( siteId: number ) => ( {
+	queryKey: [ 'site-by-slug', siteId, SITE_FIELDS, SITE_OPTIONS ],
+	queryFn: async () => {
+		try {
+			return await fetchSite( siteId );
+		} catch ( e: any ) /* eslint-disable-line @typescript-eslint/no-explicit-any */ {
+			if ( e.error === 'unknown_blog' ) {
+				throw notFound();
+			}
+			throw e;
+		}
+	},
+	meta: {
+		persist: false,
+	},
+} );
+
+export const createSiteBySlugQueryFilter = ( siteId: number ) => ( {
```

Comment:
Could you explain more about the naming `createSiteBySlugQueryFilter`? Looking at the function name, my first thought was that's related to creating a site. Maybe something like siteByIdFilter()?

Also it's a bit odd that that they query key is `site-by-slug` but it's comparing site ID.

Reply:
Rename to `createFilterForSiteSlugQuery`. The function returns a filter for the query we need, so maybe we can follow the createFilterForXXXQuery naming pattern. Any thoughts 😄